tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,547 @@
+                             spacetime_metric_ext, pi_ext, phi_ext, lapse_ext,
+                             shift_ext, inverse_spatial_metric_ext, gamma_1,
+                             gamma_2, minus_unit_normal_one_form_ext);
+
+  // Check that if the same fields are given for the interior and exterior
+  // (except that the normal vector gets multiplied by -1.0) that the
+  // numerical flux reduces to the flux
+  auto psi_normal_dot_numerical_flux = make_with_value<
+      db::item_type<::Tags::NormalDotNumericalFlux<gr::Tags::SpacetimeMetric<
+          spatial_dim, Frame::Inertial, DataVector>>>>(one, 0.0);
+  auto pi_normal_dot_numerical_flux =
+      make_with_value<db::item_type<::Tags::NormalDotNumericalFlux<
+          GeneralizedHarmonic::Tags::Pi<spatial_dim, Frame::Inertial>>>>(one,
+                                                                         0.0);
+  auto phi_normal_dot_numerical_flux =
+      make_with_value<db::item_type<::Tags::NormalDotNumericalFlux<
+          GeneralizedHarmonic::Tags::Phi<spatial_dim, Frame::Inertial>>>>(one,
+                                                                          0.0);
+  ::TestHelpers::NumericalFluxes::apply_numerical_flux(
+      flux_computer, packaged_data_int, packaged_data_int_opposite_normal,
+      make_not_null(&psi_normal_dot_numerical_flux),
+      make_not_null(&pi_normal_dot_numerical_flux),
+      make_not_null(&phi_normal_dot_numerical_flux));
+
+  ::GeneralizedHarmonic::ComputeNormalDotFluxes<spatial_dim>
+      normal_dot_flux_computer{};
+  auto psi_normal_dot_flux = make_with_value<
+      db::item_type<::Tags::NormalDotNumericalFlux<gr::Tags::SpacetimeMetric<
+          spatial_dim, Frame::Inertial, DataVector>>>>(one, 0.0);
+  auto pi_normal_dot_flux =
+      make_with_value<db::item_type<::Tags::NormalDotNumericalFlux<
+          GeneralizedHarmonic::Tags::Pi<spatial_dim, Frame::Inertial>>>>(one,
+                                                                         0.0);
+  auto phi_normal_dot_flux =
+      make_with_value<db::item_type<::Tags::NormalDotNumericalFlux<
+          GeneralizedHarmonic::Tags::Phi<spatial_dim, Frame::Inertial>>>>(one,
+                                                                          0.0);
+  normal_dot_flux_computer.apply(
+      make_not_null(&psi_normal_dot_flux), make_not_null(&pi_normal_dot_flux),
+      make_not_null(&phi_normal_dot_flux), spacetime_metric_int, pi_int,
+      phi_int, gamma_1, gamma_2, lapse_int, shift_int,
+      inverse_spatial_metric_int, unit_normal_one_form_int);
+
+  CHECK_ITERABLE_APPROX(psi_normal_dot_numerical_flux, psi_normal_dot_flux);
+  CHECK_ITERABLE_APPROX(pi_normal_dot_numerical_flux, pi_normal_dot_flux);
+  CHECK_ITERABLE_APPROX(phi_normal_dot_numerical_flux, phi_normal_dot_flux);
+}
+
+// Test GH upwind flux by comparing to Schwarzschild
+template <typename Solution>
+void test_upwind_flux_analytic(
```

Comment:
Does this test anything that the random test can not? I can't see how using specific values for the spacetime quantities, instead of random values, adds value to the test.